refactor: replace the `pofile` dependency with `pofile-ts`

swernerx opened the pull request:
Replaces the `pofile` dependency with [`pofile-ts`](https://github.com/sebastian-software/pofile-ts), a modernized fork with full TypeScript support.

**Why this change?**
- **TypeScript-first**: Native type definitions, no `@types/` package needed
- **Zero dependencies**: Smaller install footprint  
- **Modern API**: Promise-based `save()` method instead of callbacks
- **Maintained**: `pofile` last release was December 2022

**Affected packages:**
- `@lingui/format-po`
- `@lingui/format-po-gettext`
- `@lingui/cli`

**Code improvements:**
- Use exported types (`Item`, `Headers`) directly from `pofile-ts`
- Convert callback-based `po.save()` to Promise-based API
- Remove manual type definitions

**Other:**
- Add `pofile-ts` to Jest `transformIgnorePatterns` for ESM support
- Add `.lingui/` to `.gitignore` (test cache artifacts)
- Document required `yarn release:build` step in CONTRIBUTING.md

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Examples update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation (if appropriate)

timofei-iatsenko commented:
FYI https://github.com/timofei-iatsenko/pofile

timofei-iatsenko commented:
@swernerx it's not about integrating pofile into lingui, it's about hosting the package under the lingui org. So the package would be still separate, installable, but would be at full control of the lingui organization and with more trust in it from the consumers.

The problem with packages implemented by some random guy, is that, people usually don't realize how much work it would be for them after releasing something like that and burnout quite quickly leaving the package abandoned.

timofei-iatsenko commented on jest.config.js:
```diff
@@ -9,7 +9,7 @@ const tsConfigPathMapping = pathsToModuleNameMapper(
 
 const testMatch = ["**/?(*.)test.(js|ts|tsx)", "**/test/index.(js|ts|tsx)"]
-const transformIgnorePatterns = ["node_modules/(?!@messageformat)"]
+const transformIgnorePatterns = ["node_modules/(?!@messageformat|pofile-ts)"]
```
I saw that your package is distributed as dual cjs/esm, if everything setup correctly for the dual package jest should consume CJS version and this should not be needed. 

If it's still needed - it means more serious problem here.

 it means commonjs code of linguijs would require a esm package which was not possible until nowdays, but fortunately [possible today](https://socket.dev/blog/require-esm-backported-to-node-js-20). However it will bump a minimal compatible version of nodejs for users to 20.16+, 22.19+, or 24+ which i would consider a breaking change

timofei-iatsenko commented:
@swernerx meanwhile maybe you can work on these problems in your fork:

---
Multiline formatting issue: https://github.com/lingui/js-lingui/issues/2235 

Here i would suggest to have a wrap or fold option similar to gettext-parser's https://www.npmjs.com/package/gettext-parser#:~:text=with%20possible%20values-,foldLength,-is%20the%20length

> foldLength is the length at which to fold message strings into newlines (default: 76). Set to 0 or false to disable folding. 

This later might be exposed as the formatter parameter. 

Also would be nice to dig into original source and specs to understand why `pofile` creates an empty string when multiline message: 

```po
msgid ""
"Something\n"
"or something else"
```

This change might be a disruptive a bit, i would not consider it's breaking because, files generated would be still consumable  and result should not be changed, but changes here will definitely produce a diff in consumers codebases.  

---

PO translator comments must begin with # and space https://github.com/lingui/js-lingui/issues/2098

This should be easy to fix. Will also produce diff but not breaking.

swernerx commented:
> @swernerx meanwhile maybe you can work on these problems in your fork:
> 
> Multiline formatting issue: #2235
> 
> Here i would suggest to have a wrap or fold option similar to gettext-parser's https://www.npmjs.com/package/gettext-parser#:~:text=with%20possible%20values-,foldLength,-is%20the%20length
> 
> > foldLength is the length at which to fold message strings into newlines (default: 76). Set to 0 or false to disable folding.
> 
> This later might be exposed as the formatter parameter.
> 
> Also would be nice to dig into original source and specs to understand why `pofile` creates an empty string when multiline message:
> 
> ```po
> msgid ""
> "Something\n"
> "or something else"
> ```
> 
> This change might be a disruptive a bit, i would not consider it's breaking because, files generated would be still consumable and result should not be changed, but changes here will definitely produce a diff in consumers codebases.
> 
> PO translator comments must begin with # and space #2098
> 
> This should be easy to fix. Will also produce diff but not breaking.

Indeed my fork had the same issue. I fixed that and also implemented two new options to tweak it as needed for different usage models.

Fixes: https://github.com/lingui/js-lingui/pull/2379 (hopefully) ;)

timofei-iatsenko commented on packages/format-po/src/po.ts:
````diff
@@ -107,6 +108,40 @@ export type PoFormatterOptions = {
+   * msgid ""
+   * "First line\n"
+   * "Second line"
+   * ```
````
For reference the exact example with detailed description is here  
https://www.gnu.org/software/gettext/manual/html_node/More-Details.html#Further-details-on-the-PO-file-format

What i'm thinking, is that if this pattern is stated in the official docs - this should be a default, not the "compact" style, wdyt? 

I would also put a link to this doc into the flag description.

timofei-iatsenko commented:
@swernerx it's failing on the tests now. 

I'm wondering may be you can also look at the performance side of that pofile parsing? We had a discussion https://github.com/lingui/js-lingui/pull/2320#issuecomment-3244865575 some time ago which you might be interested to follow.  TL; DR; When you have a quite big catalog parsing is taking a significant time. 

I'm not sure is it something that could be improved a lot, but at least you could give it a try.
